File: Modeling/app/src/main/java/pisarev/com/modeling/interfaces/DrawMvp.java
package pisarev.com.modeling.interfaces;

import pisarev.com.modeling.mvp.model.MyData;

public interface DrawMvp {

    interface PresenterDrawViewMvp {
        void onButtonStart();

        void onButtonCycleStart();

        void onButtonSingleBlock(boolean isSingleBlockDown);

        void onButtonReset();

        void setActivity(DrawViewMvp secondView);

        void setIndex(int index);

        void setData(MyData data);
    }

    interface DrawViewMvp {
        void showFrame(String frame);

        void showAxis(String horizontalAxis, String verticalAxis);

        void showIndex(int index);
    }
}

File: Modeling/app/src/main/java/pisarev/com/modeling/interfaces/DrawMvp.kt
package pisarev.com.modeling.interfaces

import pisarev.com.modeling.mvp.model.core.MyData

interface DrawMvp {
    interface PresenterDrawViewMvp {
        fun onButtonStart()
        fun onButtonCycleStart()
        fun onButtonSingleBlock(isSingleBlockDown: Boolean)
        fun onButtonReset()
        fun setActivity(secondView: DrawViewMvp)
        fun setIndex(index: Int)
        fun setData(data: MyData)
    }

    interface DrawViewMvp {
        fun showFrame(frame: String?)
        fun showAxis(horizontalAxis: String?, verticalAxis: String?)
        fun showIndex(index: Int)
    }
}